pallets/parachain-staking/src/lib.rs

```diff
@@ -843,26 +864,44 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(n: T::BlockNumber) -> Weight {
-			let mut round = <Round<T>>::get();
-			if round.should_update(n) {
-				// mutate round
-				round.update(n);
+			let mut current_round = <Round<T>>::get();
+			if current_round.should_run_election(n) {
+				// execute all delayed nominator exits
+				Self::execute_nominator_exits(current_round.current + 1u32);
+				let election_result = Self::select_top_candidates(current_round.current + 1u32);
+				QueuedElectionResult::<T>::put(election_result);
+				// TODO: benchmark and set actual weight instead of acting like its in next block
+			}
+
+			if current_round.should_update(n) {
+				let ElectionResult {
+					collators,
+					nomination_count,
+					total_staked,
+				} = match QueuedElectionResult::<T>::take() {
+					Some(result) => result,
+					None => Self::select_top_candidates(current_round.current + 1u32),
+				};
+				let new_round = <NextRound<T>>::take().unwrap_or_else(|| {
+					current_round.update(n);
+					current_round
+				});
```

Comment:
is the branch where this config changed, does it also increment the round index?

Reply:
done now